### DISCUSSION TITLE
Ability to modify temperature in GitHub Copilot for Vim

### DISCUSSION THREAD

**Discussion by integrate-your-mind** [community/community#47993] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

I have been using the Vim version of GitHub Copilot and I am very impressed with its functionality and ability to generate code based on natural language input. However, I have noticed that I cannot modify the temperature setting in the Vim version of the extension.

For those who are not familiar with the concept, temperature refers to the level of creativity and randomness in the code generated by GitHub Copilot. It can be adjusted to generate code that is more conservative or more innovative.

As a developer, I find it important to have the ability to fine-tune the temperature setting to suit my needs and preferences. This feature is available in the VS Code version of GitHub Copilot, but unfortunately, it is not available in the Vim version.

I would like to request that the temperature setting be added to the Vim version of GitHub Copilot. This would greatly enhance the usability of the extension for Vim users, and allow us to generate code that is more tailored to our specific needs.

I have searched for similar discussions, but have not found any related to this specific issue. I would appreciate any feedback or insights from the GitHub Copilot team on this matter. Thank you.

**Comment by AsafMeizner** [comment]
its simple, dont use vim...

**Reply by integrate-your-mind** [reply]
Thank you, @PopiChan, for your response. However, I think many developers, including myself, enjoy using Vim and would appreciate having this feature available in the Vim version of GitHub Copilot. It's important to provide support and functionality for a variety of text editors to cater to different preferences within the development community.

I'm still looking forward to hearing from the GitHub Copilot team or anyone with more information on whether it's possible to add the temperature setting to the Vim extension. This feature would greatly improve the user experience for those of us who prefer using Vim.

**Reply by AsafMeizner** [reply]
I think you might be able to do that using copilot labs although I'm not sure if it's available for vim
